x-pack/plugins/stack_connectors/server/connector_types/slack/validators.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { i18n } from '@kbn/i18n';
+import { URL } from 'url';
+import type { SlackSecrets } from '../../../common/slack/types';
+
+export const validateCommonSecrets = (
+  secrets: SlackSecrets,
+  validatorServices: ValidatorServices
+) => {
+  if (!('webhookUrl' in secrets)) return;
+
+  const { configurationUtilities } = validatorServices;
+  const configuredUrl = secrets.webhookUrl;
+
+  try {
+    new URL(configuredUrl);
+  } catch (err) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.slack.configurationErrorNoHostname', {
+        defaultMessage: 'error configuring slack action: unable to parse host name from webhookUrl',
+      })
+    );
+  }
+  try {
+    configurationUtilities.ensureUriAllowed(configuredUrl);
+  } catch (allowListError) {
+    throw new Error(
+      i18n.translate('xpack.stackConnectors.slack.configurationError', {
+        defaultMessage: 'error configuring slack action: {message}',
+        values: {
+          message: allowListError.message,
+        },
+      })
+    );
+  }
+};
+
+export const validate = {
+  secrets: validateCommonSecrets,
```

Comment:
Config validation looks good.  We still need to add a 'connector' validator, as is done below in cases, with `connector: validate.connector`. 

The idea is that sometimes you need to validate that the config AND params make sense together.  We need that here, so that if they pick webhook in the config, we make sure they have message in the params, for example.

https://github.com/elastic/kibana/blob/1aa9f3293bcaca2af29173ba636c9f6caf6d984a/x-pack/plugins/stack_connectors/server/connector_types/cases_webhook/index.ts#L42-L62

Reply:
I've created a connector validation: ee91d0114399a50c0ce4ba9bb19ec42abd440316
But this one does not check parameters, but only config and secrets connection, because we do not have params argument here.